tests/contract_testcases/evm/overloaded.sol

```diff
@@ -0,0 +1,112 @@
+contract C {
+	function foo(int) public {}
+	function foo(int a, bool b) public {}
+	function foo(bool b, int a) public {}
+	function test1() public {
+		foo({a: true, b: 1});
+	}
+
+	function test2() public {
+		foo({a: 1, b: true});
+	}
+
+	function test3() public {
+		foo({a: moo, b: meh});
+	}
+}
+
+contract B is C {
+	function test1() public {
+		B.foo({a: true, b: 1});
+	}
+
+	function test2() public {
+		B.foo({a: 1, b: true});
+	}
+
+	function test3(C c) public {
+		c.foo({a: true, b: 1});
+	}
+
+	function test4(C c) public {
+		c.foo({a: 1, b: true});
+	}
+
+	function test5(C c) public {
+		c.bar({a: 1, b: true});
+	}
+
+	function test6(C c) public {
+		c.bar({a: 1, b: x});
+	}
+}
+
+// ---- Expect: diagnostics ----
+// error: 5:2-25: function 'test1' with this signature already defined
+// 	note 19:2-25: previous definition of function 'test1'
+// error: 6:3-23: cannot find overloaded function which matches signature
+// 	note 2:2-29: candidate function
+// error: 6:3-23: cannot find overloaded function which matches signature
+// 	note 3:2-39: candidate function
+// error: 6:3-23: cannot find overloaded function which matches signature
+// 	note 4:2-39: candidate function
+// error: 6:3-23: function cannot be called with named arguments as 1 of its parameters do not have names
+// 	note 2:2-26: definition of foo
+// 	note 2:2-29: candidate function
+// error: 6:11-15: conversion from bool to int256 not possible
```

Comment:
> All our conversion errors and warning say "to X from Y".

@seanyoung This is not true.

Reply:
@LucasSte as always, pedantic and not constructive.